packages/ethereum-contracts/contracts/agreements/gdav1/SuperfluidPoolPlaceholder.sol

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: AGPLv3
+// solhint-disable not-rely-on-time
+pragma solidity 0.8.19;
+
+import { BeaconProxiable } from "../../upgradability/BeaconProxiable.sol";
+
+/**
+ * @title used on first deployment (upgrade case) of GDA
+ * in order to solve the circular dependency between GDA and SuperfluidPool
+ */
+contract SuperfluidPoolPlaceholder is BeaconProxiable {
```

Comment:
You sure we have to use it instead of just using ZERO_ADDRESS?


```
nix-shell$ git grep SuperfluidPoolPlaceholder
packages/ethereum-contracts/contracts/agreements/gdav1/SuperfluidPoolPlaceholder.sol:contract SuperfluidPoolPlaceholder is BeaconProxiable {
packages/ethereum-contracts/ops-scripts/deploy-framework.js:        "SuperfluidPoolPlaceholder",
packages/ethereum-contracts/ops-scripts/deploy-framework.js:        SuperfluidPoolPlaceholder,
packages/ethereum-contracts/ops-scripts/deploy-framework.js:            SuperfluidPoolPlaceholder.new,
packages/ethereum-contracts/ops-scripts/deploy-framework.js:            "SuperfluidPoolPlaceholder.new"
```

Reply:
we need it in order to have a _compatible logic_:
```
contract SuperfluidUpgradeableBeacon is UpgradeableBeacon {
...
    function upgradeTo(address newImplementation) public override onlyOwner {
...
        if (BeaconProxiable(newImplementation).proxiableUUID() != BeaconProxiable(implementation()).proxiableUUID()) {
            revert INCOMPATIBLE_LOGIC();
        }
```

Reply:
interesting, checkout the SuperfluidDeployer code, it uses zero address, what's the difference?

Reply:
there were 2 reasons:
* don't waste gas for deploying a logic which is then immediately replaced

and more importantly:
* have all code paths blocked during the intermediate phase with the new agreement registered, but not yet properly set up (if `SuperfluidPool.initialize()` reverts, then `GeneralDistributionAgreementV1.createPool()` reverts too, thus we know for sure there's no way to use the agreement to manipulate SuperToken state as long as we're not done with the setup).

Reply:
we can remove this after upgrading all mainnets, not needed for new deployments